ergotree-ir/src/address.rs

```diff
@@ -377,13 +523,16 @@ impl AddressEncoder {
 
         let content_bytes: Vec<u8> = without_checksum[1..].to_vec(); // without head_byte
         let address_type = AddressTypePrefix::try_from(bytes[0] & 0xF_u8)?;
-        Ok(match address_type {
-            AddressTypePrefix::P2Pk => {
-                Address::P2Pk(ProveDlog::new(EcPoint::sigma_parse_bytes(&content_bytes)?))
-            }
-            AddressTypePrefix::Pay2S => Address::P2S(content_bytes),
-            // AddressTypePrefix::Pay2Sh => todo!(""),
-        })
+        match address_type {
+            AddressTypePrefix::P2Pk => Ok(Address::P2Pk(ProveDlog::new(
+                EcPoint::sigma_parse_bytes(&content_bytes)?,
+            ))),
+            AddressTypePrefix::Pay2S => Ok(Address::P2S(content_bytes)),
+            AddressTypePrefix::Pay2Sh => match <[u8; 24]>::try_from(content_bytes) {
```

Comment:
Neat match!

Reply:
Agreed, I was pretty happy when this worked!